### ISSUE TITLE
Repository & organization pages don't redirect to proper letter casing

### ISSUE THREAD

**Issue by ell1e** [go-gitea/gitea#21496]
### Description

Repository and organization pages don't redirect to the proper letter casing, for example if you visit this URL: https://try.gitea.io/blablablablab/Blatest (`Blatest`!) and then if you click issues, you'll end up at https://try.gitea.io/blablablablab/BlaTest/issues (`BlaTest`!). This is not only slightly confusing but can be a problem for search engines since the document path in URIs may or may not be case sensitive, so they may not reliably know if two different casing URIs point to the same site or not, especially if the site has dynamic contents.

The solution is, if the spelling of any repo or org URI was matched case insensitive but is *not* the canonical way of its letter casing specified in the settings, visitors should get a 301 permanent HTTP(S) redirect to correct the spelling. E.g., https://try.gitea.io/blablablablab/BlaTeST shouldn't just show the repository directly, but rather get me to the repo's proper spelling: https://try.gitea.io/blablablablab/BlaTest and the same for organization spelling too.

### Gitea Version

1.18.0+dev-587-g716fcfcf7

### Can you reproduce the bug on the Gitea demo site?

Yes

### Screenshots

![gitea-wrong-capitalization](https://user-images.githubusercontent.com/64124388/196376340-60f9c5cc-875c-4f77-a9a1-ea87a1638362.png)

### How are you running Gitea?

I'm not, I'm reproducing this on others' instances

**Comment by KN4CK3R** [comment]
Just for reference: Github does not redirect to the correct letter casing too.

**Comment by ell1e** [comment]
Yeah, they don't, very annoying. I'm happy to have moved my repositories to a Gitea-based hoster.

**Comment by yp05327** [comment]
IIRC, this issue has been fixed. But forget which PR fixed it, need some time to figure out.
